functions/Find-DbaOrphanedFile.ps1

```diff
@@ -0,0 +1,225 @@
+
+.EXAMPLE   
+Find-DbaOrphanedFile -SqlServer sqlserver2014 -Sourcedirectory 'C:\Dir1', 'C:\Dir2'
+Finds the orphaned files in the default directories but also the extra ones
+	
+
+#>
+	
+	# This is a sample. Please continue to use aliases for discoverability. Also keep the [object] type for sqlserver.
+	[CmdletBinding(SupportsShouldProcess = $true)]
+	Param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlInstance")]
+		[object]$SqlServer,
+        [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[object]$SqlCredential,
+        [parameter(Mandatory = $false, ValueFromPipeline = $true)]
+        [string[]]$Path
+	)
+	
+	BEGIN
+	{
+        
+        function Get-SqlFileStructure
+        {
+            param
+            (
+                [Parameter(Mandatory = $true, Position=1)]
+                [ValidateNotNullOrEmpty()]
+                [Alias("server")]
+                [Microsoft.SqlServer.Management.Smo.SqlSmoObject]$smoserver
+            )
+
+	        if ($smoserver.versionMajor -eq 8)
+	        {
+		        $sql = "select DB_NAME (dbid) as dbname, name, filename, CAST(mf.Size * 8 AS DECIMAL(20,2)) AS sizeKB, '' AS Drive, '' AS DestinationFolderPath, groupid from sysaltfiles"
+	        }
+	        else
+	        {
+		        $sql = "SELECT db.name AS dbname, type_desc AS FileType, mf.name, Physical_Name AS filename, CAST(mf.Size * 8 AS DECIMAL(20,2)) AS sizeKB, '' AS Drive, '' AS DestinationFolderPath FROM sys.master_files mf INNER JOIN  sys.databases db ON db.database_id = mf.database_id"
+	        }
+			
+	        $dbfiletable = $smoserver.ConnectionContext.ExecuteWithResults($sql)
+	        $ftfiletable = $dbfiletable.Tables[0].Clone()
+	        $dbfiletable.Tables[0].TableName = "data"
+			
+	        foreach ($db in $databaselist)
+	        {
+		        # Add support for Full Text Catalogs in Sql Server 2005 and below
+		        if ($server.VersionMajor -lt 10)
+		        {
+			        #$dbname = $db.name
+			        $fttable = $null = $smoserver.Databases[$database].ExecuteWithResults('sp_help_fulltext_catalogs')
+					
+			        foreach ($ftc in $fttable.Tables[0].rows)
+			        {
+				        $name = $ftc.name
+				        $physical = $ftc.Path
+				        $logical = "sysft_$name"
+				        $null = $ftfiletable.Rows.add($database, "FULLTEXT", $logical, $physical)
+			        }
+		        }
+	        }
+			
+	        $null = $dbfiletable.Tables.Add($ftfiletable)
+	        return $dbfiletable
+        }
+
+	}
+	
+	PROCESS
+	{
+        # Get the servername from the sqlinstance parameter
+        if($SqlServer.Contains('\'))
+        {
+            $servername = $SqlServer.Split('\')[0]
+        }
+        else
+        {
+            $servername = $SqlServer
+        }
+
+        # Check if the servername is the local server
+        if(($env:computername -eq $servername) -or ($env:computername -eq '127.0.0.1'))

```

Comment:
$env:computername will never equal 127.0.0.1, it should be $servername -eq '127.0.0.1'